Yoast/yoast-components
# Introduce an ErrorBoundary component and use it around the SnippetEditorFields

## Summary

* Introduce an ErrorBoundary component to be used for the Snippet Editor.

## Relevant technical choices:

- for now, the error boundary wraps the SnippetEditorFields component
- it can be changed to wrap any other components tree, if desired
- uses `a11ySpeak` to deliver an audible message to screen reader users

Design feedback needed /Cc @hedgefield 

We can't predict where we'll need to re-use this component and show errors, so we can't set arbitrary paddings / margins. For this reason, I've aligned the text to the centre and used a bottom border, as it seemed the most pragmatical solution to me. Of course, this can be changed if desired, just let me know. Personally, I'd add `moar` white space. Screenshot:

<img width="816" alt="screen shot 2018-06-19 at 17 32 52" src="https://user-images.githubusercontent.com/1682452/41609088-4aad4a70-73ea-11e8-8b81-e23205a9f06d.png">

## Test instructions
You can test the standalone version or yarn-link to the plugin:

- with Microsoft Edge on Windows: select the title field content using Ctrl+A
- with the content still selected, press any character key to trigger the Edge error
- check the error message is correctly displayed
- an audible message is sent to the ARIA live region

If you can't test with Edge, you have to change the error boundary initial state to `true`, then open the snippet editor to see the error message.

Fixes #591
